components/google_drive/sources/new-or-modified-comments/new-or-modified-comments.js

```diff
@@ -0,0 +1,185 @@
+
+    // Component was invoked by timer
+    if (event.interval_seconds) {
+      const {
+        newChannelID,
+        newPageToken,
+        expiration,
+        resourceId,
+      } = await this.googleDrive.invokedByTimer(
+        this.drive,
+        subscription,
+        this.http.endpoint,
+        channelID,
+        pageToken
+      );
+
+      this.db.set("subscription", { expiration, resourceId });
+      this.db.set("pageToken", newPageToken);
+      this.db.set("channelID", newChannelID);
+      return;
+    }
+
+    const { headers } = event;
+
+    if (!this.googleDrive.checkHeaders(headers, subscription, channelID)) {
+      return;
+    }
+
+    if (!includes(this.updateTypes, headers["x-goog-resource-state"])) {
+      console.log(
+        `Update type ${headers["x-goog-resource-state"]} not in list of updates to watch: `,
+        this.updateTypes
+      );
+      return;
+    }
+
+    // We observed false positives where a single change to a document would trigger two changes:
+    // one to "properties" and another to "content,properties". But changes to properties
+    // alone are legitimate, most users just won't want this source to emit in those cases.
+    // If x-goog-changed is _only_ set to "properties", only move on if the user set the prop
+    if (
+      !this.watchForPropertiesChanges &&
+      headers["x-goog-changed"] === "properties"
+    ) {
+      console.log(
+        "Change to properties only, which this component is set to ignore. Exiting"
+      );
+      return;
+    }
+
+    const {
+      changedFiles,
+      newStartPageToken,
+    } = await this.googleDrive.getChanges(
+      pageToken,
+      this.drive === "myDrive" ? null : this.drive
+    );
+
+    this.db.set("pageToken", newStartPageToken);
+
+    for (const file of changedFiles) {
+      const startModifiedTime = this.db.get(file.id) || null;
+      let maxModifiedTime = startModifiedTime;
+      const comments = await this.googleDrive.listComments(
+        file.id,
+        startModifiedTime
+      );
+      for (const comment of comments.comments) {
+        if (
+          !maxModifiedTime ||
+          new Date(comment.modifiedTime) > new Date(maxModifiedTime)
+        )
+          maxModifiedTime = comment.modifiedTime;
+        if (
+          startModifiedTime &&
+          new Date(comment.modifiedTime) <= new Date(startModifiedTime)
+        )
+          continue;
+
+        const eventToEmit = {
+          comment,
+          file,
+          change: {
+            state: headers["x-goog-resource-state"],
+            resourceURI: headers["x-goog-resource-uri"],
+            changed: headers["x-goog-changed"], // "Additional details about the changes. Possible values: content, parents, children, permissions"
+          },
+        };
+
+        this.$emit(eventToEmit, {
+          summary: comment.content,
+          id: headers["x-goog-message-number"],
+        });
+      }
+      this.db.set(file.id, maxModifiedTime);
```

Comment:
If we extract this into a method, we can basically replace the rest of the code in this file by extending the `new-or-modified-files` event source, correct?